crypto/src/main/scala/org/bitcoins/crypto/CryptoUtil.scala

```diff
@@ -1,154 +1,70 @@
 package org.bitcoins.crypto
 
-import java.math.BigInteger
-import java.security.MessageDigest
-
-import org.bouncycastle.crypto.digests.{RIPEMD160Digest, SHA512Digest}
-import org.bouncycastle.crypto.macs.HMac
-import org.bouncycastle.crypto.params.KeyParameter
 import org.bouncycastle.math.ec.ECPoint
-import scodec.bits.{BitVector, ByteVector}
+import scodec.bits.ByteVector
+
+import java.math.BigInteger
 
 /** Utility cryptographic functions
+  * This is a proxy for the underlying implementation of [[CryptoRuntime]]
+  * such as [[JvmCryptoRuntime]].
+  *
+  * This is necessary so that the core module doesn't need to be refactored
+  * to add support for multiple platforms, it can keep referencing CryptoUtil
   */
-trait CryptoUtil {
+trait CryptoUtil extends CryptoRuntime {
+
+  /** The underlying runtime for the specific platform we are running on */
+  private lazy val cryptoRuntime: CryptoRuntime = CryptoContext.cryptoRuntime
```

Comment:
I'm a little confused at the simultaneous is-a and has-a, why both?

Reply:
Why have a `CryptoUtil` at all if people can just call `CryptoRuntime`?

Reply:
https://github.com/bitcoin-s/bitcoin-s/pull/2658#issuecomment-778315942

If you want to see what a previous implementation looked like before doing this, see f1c9989